force-app/main/default/classes/NKS_KnowledgeBrokenLinksFinder.cls

```diff
@@ -0,0 +1,245 @@
+    }
+
+    public void finish(Database.BatchableContext bc) {
+        // toAddress: nav.kontaktsenter.fag.og.kvalitet@nav.no
+        String[] toAddresses = new List<String>{ 'sara.mohammadi@nav.no' };
+        sendEmail(finalList, toAddresses);
+    }
+
+    @TestVisible
+    private static Integer detectRedirect(HttpRequest req, HttpResponse res) {
+        Boolean redirect;
+        Integer statusCode = res.getStatusCode();
+
+        if (Test.isRunningTest()) {
+            System.debug('Running tests.');
+            return null;
+        } else {
+            if (statusCode >= 300 && statusCode <= 307 && statusCode != 306) {
+                do {
+                    redirect = false; // reset the value each time
+                    String loc = res.getHeader('Location'); // get location of the redirect
+                    if (loc == null) {
+                        redirect = false;
+                        continue;
+                    }
+                    req = new HttpRequest();
+                    req.setEndpoint(loc);
+                    req.setMethod('GET');
+                    Http http = new Http();
+                    res = http.send(req);
+                    if (statusCode != 500) {
+                        // 500 = fail
+                        if (statusCode >= 300 && statusCode <= 307 && statusCode != 306) {
+                            redirect = true;
+                        }
+                    }
+                } while (redirect && Limits.getCallouts() != Limits.getLimitCallouts());
+            }
+            return statusCode;
+        }
+    }
+
+    @TestVisible
+    private static HttpResponse calloutHttpReq(HttpRequest req, String endpoint) {
+        req.setEndpoint(endpoint);
+        req.setMethod('GET');
+        Http http = new Http();
+        HTTPResponse res = http.send(req);
+        return res;
+    }
+
+    @TestVisible
+    private static void sendEmail(Map<Id, List<String>> listToSend, List<String> toAddresses) {
+        if (listToSend.size() > 0) {
+            Messaging.SingleEmailMessage mail = new Messaging.SingleEmailMessage();
+
+            mail.setToAddresses(toAddresses);
+            //mail.setCcAddresses(ccAddresses);
+            mail.setSenderDisplayName('Døde lenker i kunnskapsbasen');
+            mail.setSubject('Rapport om døde lenker');
+            mail.setBccSender(false);
+            mail.setUseSignature(false);
+
+            // Specify the text content of the email.
+            String body = '';
+            for (Id key : listToSend.keySet()) {
+                Knowledge__kav article = [SELECT Id, Title FROM Knowledge__kav WHERE Id = :key];
+                List<String> links = listToSend.get(key);
+                body += 'Artikkel: "' + article.Title + ' " har følgende døde lenker:<br/>';
+                for (String str : links) {
+                    body += str + '<br/>';
+                }
+                body += '<br/>';
```

Comment:
Hvordan blir dette seende ut? Jeg ser ikke noe mellomrom mellom hver for iterasjon

Reply:
De skilles med break line